components/fd/get_limit_linux.go

```diff
@@ -0,0 +1,58 @@
+//go:build linux
+// +build linux
+
+package fd
+
+import (
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/prometheus/procfs"
+)
+
+const fileMaxLinux = "/proc/sys/fs/file-max"
+
+func checkFDLimitSupported() bool {
+	_, err := os.Stat(fileMaxLinux)
+	return err == nil
+}
+
+// returns the current file descriptor limit for the host, not for the current process.
+// for the current process, use syscall.Getrlimit.
+func getLimit() (uint64, error) {
+	data, err := os.ReadFile(fileMaxLinux)
+	if err != nil {
+		return 0, err
+	}
+	return strconv.ParseUint(strings.TrimSpace(string(data)), 10, 64)
+}
+
+// "process_open_fds" in prometheus collector
+// ref. https://github.com/prometheus/client_golang/blob/main/prometheus/process_collector_other.go
```

Comment:
do we need add prom license if this code is copied over

Reply:
+1. Added.